README.md

```diff
@@ -58,42 +95,235 @@ import ReactDOM from 'react-dom';
+
+Register store config to global store instance. 
+
+* Paramters
+  - namespace {string} unique name of store
+  - bindings {object} object of store config including state and actions
+* Return value
+  - {object} store instance
+
+### useStores
+
+Hooks of using multiple store.
+
+* Paramters
+  - namespaces {array} arrays of store namespace
+* Return value
+  - {array} array of stores' bindings
+
+### useStore
+
+Hooks of using single store.
+
+* Paramters
+  - namespace {string} store namespace
+* Return value
+  - {object} store's bindings
 
-ReactDOM.render(
-  <Todo />,
-  document.body
+## Advanced use
+
+### async status
+
+icestore has built in async status support in async function, it enables user to access to async functions' loading and error status without defining extra state which will make code more consise and clean.
+
+#### API
+
+* `action.loading` - flag of whether action is executing
+  - Type: {boolean}
+  - Default: false
+* `action.error` - flag of whether action has error after executing
+  - Type: {boolean}
+  - Default: false
+* `action.disableLoading` - flag of whether disable action's loading effect, if set to true, views will not rerender when loading status changed
+  - Type: {boolean}
+  - Default: false
+* `store.disableLoading` - flag of whether disable loading effect globally. Note, action's `disableLoading` flag will always takes priority when both are set
+  - Type: {boolean}
+  - Default: false
+
+#### Example
+
+```javascript
+const todos = store.useStore('todos');
+
+useEffect(() => {
+  todos.refresh();
+}, []);
+
+return (
+  <div>
+    {todos.refresh.error ? 
+      todos.refresh.loading ? (
+	    <div>
+	      loading.......
+	    </div>
+      ) : (
+        <div>
+	      error.......
+	     </div>
+      ) 
+    : (
+       <ul>
+         {dataSource.map(({ name }) => (
+           <li>{name}</li>
+         ))}
+       </ul>
+    )}
+  <Loading />
 );
 ```
+## Testing
 
-## 示例
+Because all the state and action are cleanly contained in a plain object, it's easy to write test without mocking.
 
-- Todos：[Sandbox](https://codesandbox.io/s/2017600okp)
+Example:
 
-## 反馈
+```javascript
+describe("todos", () => {
+  test("refresh data success", async () => {
+    await todos.refresh();
+    expect(todos.dataSource).toEqual([
+      {
+        name: "react"
+      },
+      {
+        name: "vue",
+        done: true
+      },
+      {
+        name: "angular"
+      }
+    ]);
+  });
+});
+```
 
-- [issues](https://github.com/alibaba/ice/issues)
+Please refer to the `todos.spec.js` file in the above sandbox for complete reference.
 
-## 参考
+## Best Practice
+
+### Never mutate state outside actions
+
+icestore enforce all the mutations to state can only ouccr in action methods. It will throw error when mutate state directly outside actions(eg. in view component). It will cause your mutation logic hard to trace and imposible to test if let mutation scatterd around view component other than store.
```

Comment:
icestore enforces all the mutations to the state to occur only in action methods. An error would be thrown if the state is mutated outside actions (e.g. in the view component). 

The reason is that the mutation logic would be hard to trace and impossible to test as there might be unpredictable changes made to the view components as a result of mutations outside actions.